### PR TITLE
In data selection widget, only display label (not also name)

### PR DESCRIPTION
**Pull request by wpfff**
avoids redundancy and makes the widget content more legible.

### CODE REVIEW

File: plottr/gui/data_display.py [hunk 1]
```diff
@@ -32,14 +32,13 @@ def __init__(self, parent: Optional[QtWidgets.QWidget] = None,
...
-        deps = "("
+        label = f"{self._dataStructure.label(name)}"
+        deps = ""
         for i, d in enumerate(self._dataStructure.axes(name)):
             if i > 0:
                 deps += ", "
             axlabel = self._dataStructure.label(d)
-            deps += f"{d} [{axlabel}]"
-        deps += ")"
+            deps += f"{axlabel}"
```
Comment (jenshnielsen) [inline plottr/gui/data_display.py]: Personally I find it cleaner to write this with a list comprehension and and join.

```python
        axlabels = [self._dataStructure.label(d) for d in
                    self._dataStructure.axes(name)]
        deps = ",".join(axlabels)
```

Avoiding the for loop and the if statement

**Review by jenshnielsen — APPROVED** [review]

File: plottr/gui/data_display.py (hunk 1 above)
Comment (wpfff) [reply plottr/gui/data_display.py]: agreed -- i changed that.

Comment (wpfff) [reply plottr/gui/data_display.py]: hmmm, now there's a mypy error that seems to come from the list comprehension. how do you fix that without reverting to the previous verbose code?

Comment (jenshnielsen) [reply plottr/gui/data_display.py]: So the problem is that label many return `None` so axlabel is a list of optional str. I have fixed that by just converting any Nones to str

**Pull request merged**